app/controllers/listing_endorsements_controller.rb

```diff
@@ -0,0 +1,25 @@
+class ListingEndorsementsController < ApplicationController
+  before_action :raise_suspended, only: %i[create update]
+  before_action :authenticate_user!, only: %i[create update]
+  after_action :verify_authorized, only: %i[update]
+
+  def create
+    @endorsement = ListingEndorsement.create(content: params[:content],
+                                             user_id: current_user.id,
+                                             classified_listing_id: params[:classified_listing_id])
+    if @endorsement.persisted?
+      Notification.send_new_endorsement_notifications_without_delay(@endorsement)
```

Comment:
https://github.com/forem/forem/pull/9553/files?file-filters%5B%5D=.rb#r471399058

Yes it returns JSON but the JSON is generated from the notification service object after extracting that from the endorsement object sent